boards/Kconfig.v2

```diff
@@ -15,4 +16,24 @@ config BOARD_$(BOARD_FULL_STRING)
 	help
 	  Kconfig symbol identifying the board including full board identifier.
 
+config BOARD_QUALIFIERS
+	string
+	default "$(BOARD_QUALIFIERS)"
+	help
+	  Contains the qualifiers of the board being used without the name of the board itself.
+
+	  For example, if building for ``nrf5340dk/nrf5340/cpuapp`` then this will contain the
+	  value ``nrf5340/cpuapp``.
+
+config BOARD_TARGET
+	string
+	default "$(BOARD)@$(BOARD_REVISION)$(BOARD_IDENTIFIER)" if "$(BOARD_REVISION)" != ""
+	default "$(BOARD)$(BOARD_IDENTIFIER)"
+	help
+	  Contains the board target (full string including name, revision, soc, cluster and
+	  variant) of the board being used.
+
+	  For example, if building for ``nrf5340dk/nrf5340/cpuapp`` then this will contain the
+	  value ``nrf5340dk/nrf5340/cpuapp``.
```

Comment:
Please make this generally available together with other `config BOARD` in this Kconfig file:
https://github.com/zephyrproject-rtos/zephyr/blob/4b8d3dbed8a19a28a37b3fb5b46787ddb14d98b0/boards/Kconfig#L3-L21

Reply:
Note that the initial reason for the `Kconfig.v1` and `Kconfig.v2` split is because v1 boards (and SoCs) are using `choice` selections which prevents boards from using select and instead relies on a `CONFIG_BOARD_PLANK=y` and `CONFIG_SOC_FOO=y` in the `_defconfig`.

Therefore we needed to split into two trees, to remove the choice.

**Not** because we needed to  separated new Kconfigs from old Kconfigs.

See also: https://github.com/zephyrproject-rtos/zephyr/blob/b839f411613e500a8d20592d40709d1317feadca/boards/Kconfig.v1#L5-L6
https://github.com/zephyrproject-rtos/zephyr/blob/b839f411613e500a8d20592d40709d1317feadca/soc/Kconfig.v1.choice#L5-L6